snapshot-labs/sx-ui
# feat: add evmSlotValue strategy support

### Summary

Depends on: https://github.com/snapshot-labs/sx.js/pull/243
Depends on: https://github.com/snapshot-labs/mana/pull/125
Depends on: https://github.com/snapshot-labs/sx-api/pull/206

This PR adds support for evmSlotValue strategy.

Closes: https://github.com/snapshot-labs/sx-ui/issues/827

### How to test

Set up all dependencies first and make sure you are using local mana and sx-api. Check mana/sx-api test plans for more details. There is lots of dependencies so it's a bit cumbersome to set up right now.

1. Create new space on Starknet (goerli) - mainnet is not supported by Herodotus using EVM Slot value strategy. You can use this token (`0xd96844c9b21cb6ccf2c236257c7fc703e43ba071` + slot index `0` as you can mint it yourself).
2. Create new proposal.
3. Check mana logs for internalId and webhook URL (will be automated on deployment).
4. Check `https://api.herodotus.cloud/batch-query-status?apiKey=API_KEY&batchQueryId=INTERNAL_ID_RETURNED_BY_MANA` until status is `DONE`.
5. Open webhook URL.
6. Refresh page, you should have VP.
7. You should be able to vote.
